index.d.ts

```diff
@@ -2,6 +2,28 @@
 import File = require('vinyl');
 import * as globStream from 'glob-stream';
 import * as through from 'through2';
+import {EventEmitter} from 'events';
+
+declare interface WatchOptions {
+    interval?: number;
+    debounceDelay?: number;
+    cwd?: string;
+    maxListeners?: Function;
+}
+
+declare interface OutEvent {
+    type: any;
+    path: any;
+    old: any;
+}
+
+/**
+ * watches all files in the glob and runs the given callback as soon as a change to one of the watched files occurs.
+ */
+export function watch(globs: string | Array<string>): EventEmitter;
+export function watch(globs: string | Array<string>, cb: (outEvt: OutEvent) => void): EventEmitter;
+export function watch(globs: string | Array<string>, opt: WatchOptions): EventEmitter;
+export function watch(globs: string | Array<string>, opt: WatchOptions, cb: (outEvt: OutEvent) => void): EventEmitter;
```

Comment:
I could do that but to be honest I don't really see a reason for it since with optional parameters you would basically introduce implicit overrides within explicit overrides. Doesn't that feel a bit odd to you?